--- a/src/eim.cpp
+++ b/src/eim.cpp
@@ -462,7 +462,7 @@
     if (fp)
         fclose(fp);
 
-    bool result = ime_pinyin::im_open_decoder(DATADIR "/googlepinyin/dict_pinyin.dat", userDict);
+    bool result = ime_pinyin::im_open_decoder(LIBDIR "/googlepinyin/data/dict_pinyin.dat", userDict);
 
     if (userDict)
         free(userDict);

--- fcitx-googlepinyin-0.1.3.orig/CMakeLists.txt
+++ fcitx-googlepinyin-0.1.3/CMakeLists.txt
@@ -16,8 +16,9 @@ configure_file(
 add_custom_target(uninstall
     COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake)

+add_subdirectory(src)
 if(NOT DEFINED LIB_INSTALL_DIR)
-    set(LIB_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/lib)
+    set(LIB_INSTALL_DIR ${GOOGLEPINYIN_LIBDIR})
 endif()

 include(FindPkgConfig)
@@ -34,8 +35,5 @@ set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--as-
 set(CMAKE_MODULE_LINKER_FLAGS "-Wl,--as-needed ${CMAKE_MODULE_LINKER_FLAGS}")
 configure_file(config.h.in config.h)

-set(libdir ${LIB_INSTALL_DIR})
-
-add_subdirectory(src)
 add_subdirectory(data)
 add_subdirectory(po)
